pom.xml

```diff
@@ -0,0 +1,351 @@
+                            <!-- Prevent `gpg` from using pinentry programs -->
+                            <gpgArguments>
+                                <arg>--pinentry-mode</arg>
+                                <arg>loopback</arg>
+                            </gpgArguments>
+                        </configuration>
+                    </plugin>
+                    <plugin>
+                        <groupId>org.sonatype.plugins</groupId>
+                        <artifactId>nexus-staging-maven-plugin</artifactId>
+                        <extensions>true</extensions>
+                        <configuration>
+                            <serverId>ossrh</serverId>
+                            <nexusUrl>https://oss.sonatype.org/</nexusUrl>
+                            <autoReleaseAfterClose>true</autoReleaseAfterClose>
+                            <autoDropAfterRelease>true</autoDropAfterRelease>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+
+        <!-- Enables Docker image(s) build / deploy -->
+        <profile>
+            <id>docker</id>
+            <activation>
+                <property>
+                    <name>docker</name>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>com.spotify</groupId>
+                        <artifactId>dockerfile-maven-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>tag-version</id>
+                                <goals>
+                                    <goal>build</goal>
+                                    <goal>push</goal>
+                                </goals>
+                                <configuration>
+                                    <tag>${dockerfile.versionTagName}</tag>
+                                </configuration>
+                            </execution>
+                            <execution>
+                                <id>tag-latest</id>
+                                <goals>
+                                    <goal>tag</goal>
+                                    <goal>push</goal>
+                                </goals>
+                                <configuration>
+                                    <tag>${dockerfile.latestTagName}</tag>
+                                </configuration>
+                            </execution>
+                        </executions>
+                        <configuration>
+                            <skip>true</skip>
+                            <!-- Dockerfile plugin requires <repository> to be non-empty even if <skip> is true -->
+                            <repository>${dockerfile.repository}</repository>
+                            <buildArgs combine.children="append">
```

Comment:
I propose adding this to the `buildArgs`

```xml
<TARGET_NAME>${project.build.finalName}</TARGET_NAME>
```

- it is general enough
- we always want to use the build output
- we do not need to think about changing the name of the build
- we can move docker files between projects more easily

The tag name (`TARGET_NAME`) is not something I strongly endorse. It is a first name that came to mind. Feel free to change.

Reply:
I've added the following with the names corresponding to the Maven properties:
```xml
<PROJECT_NAME>${project.name}</PROJECT_NAME>
<PROJECT_GROUP_ID>${project.groupId}</PROJECT_GROUP_ID>
<PROJECT_ARTIFACT_ID>${project.artifactId}</PROJECT_ARTIFACT_ID>
<PROJECT_VERSION>${project.version}</PROJECT_VERSION>
<PROJECT_BASEDIR>${project.basedir}</PROJECT_BASEDIR>
<PROJECT_BUILD_DIRECTORY>${project.build.directory}</PROJECT_BUILD_DIRECTORY>
<PROJECT_BUILD_FINAL_NAME>${project.build.finalName}</PROJECT_BUILD_FINAL_NAME>
```